firefox-ios/Client/Frontend/PasswordGenerator/PasswordGeneratorMiddleware.swift

```diff
@@ -69,15 +75,18 @@ final class PasswordGeneratorMiddleware {
     }
 
     private func generateNewPassword(frame: WKFrameInfo, completion: @escaping (String) -> Void) {
-        let jsFunctionCall = "window.__firefox__.logins.generatePassword()"
-        frame.webView?.evaluateJavascriptInDefaultContentWorld(jsFunctionCall, frame) { (result, error) in
-            if let error = error {
-                self.logger.log("JavaScript evaluation error",
-                                level: .warning,
-                                category: .webview,
-                                description: "\(error.localizedDescription)")
-            } else if let result = result as? String {
-                completion(result)
+        Task {
```

Comment:
I'm not sure if the task block causes this issue, but I remember reading in the Redux wiki that actions should be dispatched from a single thread. I believe using the task block here would result in our action being dispatched in a different thread than all other redux actions. Source: https://github.com/mozilla-mobile/firefox-ios/wiki/Redux#architecture

Reply:
Good catch. I am not sure if it applies either but it's worth verifying. @OrlaM is this a problem ? If it is I think we can wrap some code in `DispatchQueue.main.async` but that's ugly 🙈

Reply:
Or maybe we can preload the data somehow before doing anything 🤔

Reply:
Fixed in https://github.com/mozilla-mobile/firefox-ios/pull/22929/commits/8fe490e01cb13de7c0ddb39a5b083119011a1788 now we preload data on init and cache it in static var so on theory we should only load the data once.